core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -110,7 +114,30 @@ TableScanContext useSnapshotId(Long scanSnapshotId) {
         toSnapshotId,
         planExecutor,
         fromSnapshotInclusive,
-        metricsReporter);
+        metricsReporter,
+        branch);
+  }
+
+  String branch() {
+    return branch;
+  }
+
+  TableScanContext useBranch(String newBranch) {
```

Comment:
I simplified `TableScanContext` a while ago in https://github.com/apache/iceberg/pull/5985 mainly because it's painful to add a single new field (because it requires adjusting all constructor calls). I wonder if it's worth re-visiting #5985 now

Reply:
Hey I have just merged #5985, could you rebase based on that? @hililiwei

Reply:
once you rebase, all you need to add to `TableScanContext` is:

```
@Nullable
public abstract String ref();
```

and then a `useRef()` shortcut method if required
```
TableScanContext useRef(String ref) {
  return ImmutableTableScanContext.builder().from(this).ref(ref).build();
}
```